pkg/compact/compact.go

```diff
@@ -908,7 +918,7 @@ func (c *BucketCompactor) Compact(ctx context.Context) error {
 
 		level.Info(c.logger).Log("msg", "start of compaction")
 
-		groups, err := c.sy.Groups()
+		groups, err := c.sy.Groups(c.acceptMalformedIndex)
```

Comment:
Can we put to group field as well? Essentially if it is option like this then it suggests that we change this dynamically, which we don't. Also bool argument in function is discouraged ):

Reply:
I looked at that this morning, which would mean I need to modify the Syncer to support this flag rather than BucketCompactor.  We'd pass in the command line option here:

https://github.com/jjneely/thanos/blob/jjneely/accept-malformed-index/cmd/thanos/compact.go#L170

rather than here:

https://github.com/jjneely/thanos/blob/jjneely/accept-malformed-index/cmd/thanos/compact.go#L200

That seemed like not a straight forward path.  Then I found that compaction Groups were created by the Syncer rather than the BlockCompactor.

Reply:
Ok, changed.